forge/lang/expander.rkt

```diff
@@ -724,6 +748,41 @@
        (syntax/loc stx (begin block.test-decls ...))
        (syntax/loc stx (begin)))]))
 
+
+; PropertyWhereDecl : PROPERTY-TOK Name OF-TOK Name Expr WHERE-TOK TEST-CONSTRUCT*
+(define-syntax (PropertyWhereDecl stx)
+  (syntax-parse stx
+  [((~literal PropertyWhereDecl) 
+                              (~and constraint_type (~or "overconstraint" "underconstraint"))  
+                              prop_name:NameClass
```

Comment:
All the `_` should be `-` to match style with the rest of the codebase.

`_` is more work to type, too